types/phoenix_live_view/index.d.ts

```diff
@@ -41,6 +43,13 @@ export interface DomOptions {
 
 export type ViewLogger = (view: View, kind: string, msg: any, obj: any) => void;
 
+export type EventCallback =
+    ((e: KeyboardEvent, el: Element) => Record<string, unknown>) |
+    ((e: MouseEvent, el: Element) => Record<string, unknown>) |
+    ((e: FocusEvent, el: Element) => Record<string, unknown>)
+
+export type Metadata = Record<string, EventCallback>
+
 export interface SocketOptions {
     longPollFallbackMs?: number | undefined;
```

Comment:
I missed that yesterday, but `prop?: SomeType | undefined` is redundant.
Optional means it's already maybe undefined, so there's no need to put both